### ISSUE TITLE
how to force transitive dependencies for groovy plugin's dependency?

### ISSUE THREAD

**Issue by cxjava** [gradle/gradle#22925]
### Expected Behavior

Customizing resolution of a dependency directly can works fine:

```
configurations.all {
    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
        def requested = details.requested
        if (requested.group == 'org.apache.ivy' && requested.name == 'ivy') {
            details.useVersion '2.5.1'
        }
    }
}
```

### Current Behavior

transitive dependencies not work.

### Context

when we upgrade to spring boot 3.0.0, the default groovy version change to 4.0.6(extend from spring boot bom file)

the dependency flow like: org.apache.groovy:groovy:4.0.6 -> org.apache.ivy:ivy:2.5.0(blocked by own nexus, can't release it.)

but the sub module ivy have the vulnerability, our own nexus will blocked it. When run the build task, it will failed.

We need to define newest version 2.5.1 to overwrite org.apache.ivy:ivy:2.5.0.

We try many ways to overwrite it, but failed, seems this is the **groovy plugin runtime dependency**, can't overwrite it?

Does anyone know how to fix this problem?

### Steps to Reproduce 

option 1
```
testImplementation('org.apache.groovy:groovy:4.0.6') {
    exclude group: 'org.apache.ivy', module: 'ivy'
}
testImplementation 'org.apache.ivy:ivy:2.5.1'
```

option 2
```
configurations.all {
    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
        def requested = details.requested
        if (requested.group == 'org.apache.ivy' && requested.name == 'ivy') {
            details.useVersion '2.5.1'
        }
    }
}
```

option 3
```
 testImplementation('org.apache.groovy:groovy:4.0.6') {
    exclude group: 'org.apache.ivy', module: 'ivy'
}
testImplementation('org.apache.ivy:ivy:2.5.1') {
    force = true
}
```

option 4
```
configurations.all {
    resolutionStrategy {
        force 'org.apache.ivy:ivy:2.5.1'
    }
}
```

### Your Environment

Build scan URL:

``` log
2022-12-01T09:33:00.700+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-12-01T09:33:00.701+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
2022-12-01T09:33:00.702+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-12-01T09:33:00.703+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
2022-12-01T09:33:00.703+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':compileIsolationTestGroovy'.
2022-12-01T09:33:00.703+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Could not resolve all files for configuration ':detachedConfiguration66'.
2022-12-01T09:33:00.703+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]    > Could not download ivy-2.5.0.jar (org.apache.ivy:ivy:2.5.0)
2022-12-01T09:33:00.703+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]       > Could not get resource 'https://xxx.cloud/repository/maven-central/org/apache/ivy/ivy/2.5.0/ivy-2.5.0.jar'.
2022-12-01T09:33:00.704+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]          > Could not HEAD 'https://xxx.cloud/repository/maven-central/org/apache/ivy/ivy/2.5.0/ivy-2.5.0.jar'. Received status code 403 from server: -------------------->>> REQUESTED ITEM IS QUARANTINED -------------------->>> FOR DETAILS SEE ------>>> https://xxx.cloud/ui/links/repositories/quarantinedComponent/ccccccccc <<<------
2022-12-01T09:33:00.704+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-12-01T09:33:00.704+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
2022-12-01T09:33:00.704+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Run with --stacktrace option to get the stack trace.
2022-12-01T09:33:00.704+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Run with --scan to get full insights.
2022-12-01T09:33:00.704+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-12-01T09:33:00.705+0800 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org
2022-12-01T09:33:00.705+0800 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] 
2022-12-01T09:33:00.705+0800 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] BUILD FAILED in 9s
```

gradle example:

<details>
  <summary>build.gradle </summary>
  
```
plugins {
    id 'java'
    id 'groovy'
    id 'org.springframework.boot' version '3.0.0'
    id 'io.spring.dependency-management' version '1.1.0'
}

group = 'com.example'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = '17'

// option 2
//configurations.all {
//    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
//        def requested = details.requested
//        if (requested.group == 'org.apache.ivy' && requested.name == 'ivy') {
//            details.useVersion '2.5.1'
//        }
//    }
//}
// option 2
// option 4
//configurations.all {
//    resolutionStrategy {
//        force 'org.apache.ivy:ivy:2.5.1'
//    }
//}
// option 4

dependencies {
    implementation 'org.springframework.boot:spring-boot-starter-web'
    implementation 'org.springframework.boot:spring-boot-starter'
    testImplementation 'org.springframework.boot:spring-boot-starter-test'
    testImplementation 'org.apache.groovy:groovy:4.0.6'
    // option 1
//    testImplementation('org.apache.groovy:groovy:4.0.6') {
//        exclude group: 'org.apache.ivy', module: 'ivy'
//    }
//    testImplementation 'org.apache.ivy:ivy:2.5.1'
    // option 1

    // option 3
//    testImplementation('org.apache.groovy:groovy:4.0.6') {
//        exclude group: 'org.apache.ivy', module: 'ivy'
//    }
//    testImplementation('org.apache.ivy:ivy:2.5.1') {
//        force = true
//    }
    // option 3

    testImplementation 'org.spockframework:spock-core:2.3-groovy-4.0'
    testImplementation 'org.spockframework:spock-spring:2.3-groovy-4.0'
}
sourceSets {
    isolationTest {
        groovy.srcDir file('src/isolation-test/groovy')
        resources.srcDir file('src/isolation-test/resources')
        compileClasspath += sourceSets.main.output
        runtimeClasspath += sourceSets.main.output
    }
}

configurations {
    isolationTestImplementation.extendsFrom testImplementation
    isolationTestRuntimeOnly.extendsFrom testRuntimeOnly
}

task isolationTest(type: Test) {
    testClassesDirs = sourceSets.isolationTest.output.classesDirs
    classpath = sourceSets.isolationTest.runtimeClasspath
    useJUnitPlatform()
}

isolationTest.mustRunAfter test

tasks.named('test') {
    useJUnitPlatform()
}

repositories {
    for (repo in ['maven-central', 'maven-flcit', 'maven-hosted', 'maven-public', 'gradle-plugins']) {
        maven {
            credentials {
                username "admin"
                password "xxx"
            }
            url "https://xxxx.cloud/repository/${repo}/"
        }
    }
}

//repositories {
//    mavenCentral()
//}

```

</details>

<details>
  <summary>./gradlew dependencies </summary>
  
```
> Task :dependencies

------------------------------------------------------------
Root project 'demo'
------------------------------------------------------------

annotationProcessor - Annotation processors and their dependencies for source set 'main'.
No dependencies

apiElements - API elements for main. (n)
No dependencies

archives - Configuration for archive artifacts. (n)
No dependencies

bootArchives - Configuration for Spring Boot archive artifacts. (n)
No dependencies

compileClasspath - Compile classpath for source set 'main'.
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
\--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)

compileOnly - Compile only dependencies for source set 'main'. (n)
No dependencies

default - Configuration for default artifacts. (n)
No dependencies

developmentOnly - Configuration for development-only dependencies such as Spring Boot's DevTools.
No dependencies

implementation - Implementation only dependencies for source set 'main'. (n)
+--- org.springframework.boot:spring-boot-starter-web (n)
\--- org.springframework.boot:spring-boot-starter (n)

isolationTestAnnotationProcessor - Annotation processors and their dependencies for source set 'isolation test'.
No dependencies

isolationTestCompileClasspath - Compile classpath for source set 'isolation test'.
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
+--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)
+--- org.springframework.boot:spring-boot-starter-test -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test:3.0.0
|    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test-autoconfigure:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-test:3.0.0 (*)
|    |    \--- org.springframework.boot:spring-boot-autoconfigure:3.0.0 (*)
|    +--- com.jayway.jsonpath:json-path:2.7.0
|    |    +--- net.minidev:json-smart:2.4.7 -> 2.4.8
|    |    |    \--- net.minidev:accessors-smart:2.4.8
|    |    |         \--- org.ow2.asm:asm:9.1
|    |    \--- org.slf4j:slf4j-api:1.7.33 -> 2.0.4
|    +--- jakarta.xml.bind:jakarta.xml.bind-api:4.0.0
|    |    \--- jakarta.activation:jakarta.activation-api:2.1.0
|    +--- org.assertj:assertj-core:3.23.1
|    |    \--- net.bytebuddy:byte-buddy:1.12.10 -> 1.12.19
|    +--- org.hamcrest:hamcrest:2.2
|    +--- org.junit.jupiter:junit-jupiter:5.9.1
|    |    +--- org.junit:junit-bom:5.9.1
|    |    |    +--- org.junit.jupiter:junit-jupiter:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-params:5.9.1 (c)
|    |    |    +--- org.junit.platform:junit-platform-engine:1.9.1 (c)
|    |    |    \--- org.junit.platform:junit-platform-commons:1.9.1 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1
|    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    +--- org.opentest4j:opentest4j:1.2.0
|    |    |    +--- org.junit.platform:junit-platform-commons:1.9.1
|    |    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    |    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    |    \--- org.junit.jupiter:junit-jupiter-params:5.9.1
|    |         +--- org.junit:junit-bom:5.9.1 (*)
|    |         +--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    |         \--- org.apiguardian:apiguardian-api:1.1.2
|    +--- org.mockito:mockito-core:4.8.1
|    |    +--- net.bytebuddy:byte-buddy:1.12.16 -> 1.12.19
|    |    \--- net.bytebuddy:byte-buddy-agent:1.12.16 -> 1.12.19
|    +--- org.mockito:mockito-junit-jupiter:4.8.1
|    |    \--- org.mockito:mockito-core:4.8.1 (*)
|    +--- org.skyscreamer:jsonassert:1.5.1
|    |    \--- com.vaadin.external.google:android-json:0.0.20131108.vaadin1
|    +--- org.springframework:spring-core:6.0.2 (*)
|    +--- org.springframework:spring-test:6.0.2
|    |    \--- org.springframework:spring-core:6.0.2 (*)
|    \--- org.xmlunit:xmlunit-core:2.9.0
+--- org.apache.groovy:groovy:4.0.6
|    \--- org.apache.groovy:groovy-bom:4.0.6
|         \--- org.apache.groovy:groovy:4.0.6 (c)
+--- org.apache.ivy:ivy:2.5.1
+--- org.spockframework:spock-core:2.3-groovy-4.0
|    +--- org.apache.groovy:groovy:4.0.4 -> 4.0.6 (*)
|    +--- org.junit:junit-bom:5.9.0 -> 5.9.1 (*)
|    +--- org.junit.platform:junit-platform-engine -> 1.9.1
|    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    +--- org.opentest4j:opentest4j:1.2.0
|    |    +--- org.junit.platform:junit-platform-commons:1.9.1 (*)
|    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    \--- org.hamcrest:hamcrest:2.2
\--- org.spockframework:spock-spring:2.3-groovy-4.0
     \--- org.spockframework:spock-core:2.3-groovy-4.0 (*)

isolationTestCompileOnly - Compile only dependencies for source set 'isolation test'. (n)
No dependencies

isolationTestImplementation - Implementation only dependencies for source set 'isolation test'. (n)
No dependencies

isolationTestRuntimeClasspath - Runtime classpath of source set 'isolation test'.
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
+--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)
+--- org.springframework.boot:spring-boot-starter-test -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test:3.0.0
|    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test-autoconfigure:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-test:3.0.0 (*)
|    |    \--- org.springframework.boot:spring-boot-autoconfigure:3.0.0 (*)
|    +--- com.jayway.jsonpath:json-path:2.7.0
|    |    +--- net.minidev:json-smart:2.4.7 -> 2.4.8
|    |    |    \--- net.minidev:accessors-smart:2.4.8
|    |    |         \--- org.ow2.asm:asm:9.1
|    |    \--- org.slf4j:slf4j-api:1.7.33 -> 2.0.4
|    +--- jakarta.xml.bind:jakarta.xml.bind-api:4.0.0
|    |    \--- jakarta.activation:jakarta.activation-api:2.1.0
|    +--- org.assertj:assertj-core:3.23.1
|    |    \--- net.bytebuddy:byte-buddy:1.12.10 -> 1.12.19
|    +--- org.hamcrest:hamcrest:2.2
|    +--- org.junit.jupiter:junit-jupiter:5.9.1
|    |    +--- org.junit:junit-bom:5.9.1
|    |    |    +--- org.junit.jupiter:junit-jupiter:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-engine:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-params:5.9.1 (c)
|    |    |    +--- org.junit.platform:junit-platform-engine:1.9.1 (c)
|    |    |    \--- org.junit.platform:junit-platform-commons:1.9.1 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1
|    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    +--- org.opentest4j:opentest4j:1.2.0
|    |    |    \--- org.junit.platform:junit-platform-commons:1.9.1
|    |    |         \--- org.junit:junit-bom:5.9.1 (*)
|    |    +--- org.junit.jupiter:junit-jupiter-params:5.9.1
|    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    \--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    |    \--- org.junit.jupiter:junit-jupiter-engine:5.9.1
|    |         +--- org.junit:junit-bom:5.9.1 (*)
|    |         +--- org.junit.platform:junit-platform-engine:1.9.1
|    |         |    +--- org.junit:junit-bom:5.9.1 (*)
|    |         |    +--- org.opentest4j:opentest4j:1.2.0
|    |         |    \--- org.junit.platform:junit-platform-commons:1.9.1 (*)
|    |         \--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    +--- org.mockito:mockito-core:4.8.1
|    |    +--- net.bytebuddy:byte-buddy:1.12.16 -> 1.12.19
|    |    +--- net.bytebuddy:byte-buddy-agent:1.12.16 -> 1.12.19
|    |    \--- org.objenesis:objenesis:3.2
|    +--- org.mockito:mockito-junit-jupiter:4.8.1
|    |    +--- org.mockito:mockito-core:4.8.1 (*)
|    |    \--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    +--- org.skyscreamer:jsonassert:1.5.1
|    |    \--- com.vaadin.external.google:android-json:0.0.20131108.vaadin1
|    +--- org.springframework:spring-core:6.0.2 (*)
|    +--- org.springframework:spring-test:6.0.2
|    |    \--- org.springframework:spring-core:6.0.2 (*)
|    \--- org.xmlunit:xmlunit-core:2.9.0
+--- org.apache.groovy:groovy:4.0.6
|    \--- org.apache.groovy:groovy-bom:4.0.6
|         \--- org.apache.groovy:groovy:4.0.6 (c)
+--- org.apache.ivy:ivy:2.5.1
+--- org.spockframework:spock-core:2.3-groovy-4.0
|    +--- org.apache.groovy:groovy:4.0.4 -> 4.0.6 (*)
|    +--- org.junit:junit-bom:5.9.0 -> 5.9.1 (*)
|    +--- org.junit.platform:junit-platform-engine -> 1.9.1 (*)
|    \--- org.hamcrest:hamcrest:2.2
\--- org.spockframework:spock-spring:2.3-groovy-4.0
     +--- org.apache.groovy:groovy:4.0.4 -> 4.0.6 (*)
     \--- org.spockframework:spock-core:2.3-groovy-4.0 (*)

isolationTestRuntimeOnly - Runtime only dependencies for source set 'isolation test'. (n)
No dependencies

mainSourceElements - List of source directories contained in the Main SourceSet. (n)
No dependencies

productionRuntimeClasspath
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
\--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)

runtimeClasspath - Runtime classpath of source set 'main'.
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
\--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)

runtimeElements - Elements of runtime for main. (n)
No dependencies

runtimeOnly - Runtime only dependencies for source set 'main'. (n)
No dependencies

testAnnotationProcessor - Annotation processors and their dependencies for source set 'test'.
No dependencies

testCompileClasspath - Compile classpath for source set 'test'.
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
+--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)
+--- org.springframework.boot:spring-boot-starter-test -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test:3.0.0
|    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test-autoconfigure:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-test:3.0.0 (*)
|    |    \--- org.springframework.boot:spring-boot-autoconfigure:3.0.0 (*)
|    +--- com.jayway.jsonpath:json-path:2.7.0
|    |    +--- net.minidev:json-smart:2.4.7 -> 2.4.8
|    |    |    \--- net.minidev:accessors-smart:2.4.8
|    |    |         \--- org.ow2.asm:asm:9.1
|    |    \--- org.slf4j:slf4j-api:1.7.33 -> 2.0.4
|    +--- jakarta.xml.bind:jakarta.xml.bind-api:4.0.0
|    |    \--- jakarta.activation:jakarta.activation-api:2.1.0
|    +--- org.assertj:assertj-core:3.23.1
|    |    \--- net.bytebuddy:byte-buddy:1.12.10 -> 1.12.19
|    +--- org.hamcrest:hamcrest:2.2
|    +--- org.junit.jupiter:junit-jupiter:5.9.1
|    |    +--- org.junit:junit-bom:5.9.1
|    |    |    +--- org.junit.jupiter:junit-jupiter:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-params:5.9.1 (c)
|    |    |    +--- org.junit.platform:junit-platform-engine:1.9.1 (c)
|    |    |    \--- org.junit.platform:junit-platform-commons:1.9.1 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1
|    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    +--- org.opentest4j:opentest4j:1.2.0
|    |    |    +--- org.junit.platform:junit-platform-commons:1.9.1
|    |    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    |    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    |    \--- org.junit.jupiter:junit-jupiter-params:5.9.1
|    |         +--- org.junit:junit-bom:5.9.1 (*)
|    |         +--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    |         \--- org.apiguardian:apiguardian-api:1.1.2
|    +--- org.mockito:mockito-core:4.8.1
|    |    +--- net.bytebuddy:byte-buddy:1.12.16 -> 1.12.19
|    |    \--- net.bytebuddy:byte-buddy-agent:1.12.16 -> 1.12.19
|    +--- org.mockito:mockito-junit-jupiter:4.8.1
|    |    \--- org.mockito:mockito-core:4.8.1 (*)
|    +--- org.skyscreamer:jsonassert:1.5.1
|    |    \--- com.vaadin.external.google:android-json:0.0.20131108.vaadin1
|    +--- org.springframework:spring-core:6.0.2 (*)
|    +--- org.springframework:spring-test:6.0.2
|    |    \--- org.springframework:spring-core:6.0.2 (*)
|    \--- org.xmlunit:xmlunit-core:2.9.0
+--- org.apache.groovy:groovy:4.0.6
|    \--- org.apache.groovy:groovy-bom:4.0.6
|         \--- org.apache.groovy:groovy:4.0.6 (c)
+--- org.apache.ivy:ivy:2.5.1
+--- org.spockframework:spock-core:2.3-groovy-4.0
|    +--- org.apache.groovy:groovy:4.0.4 -> 4.0.6 (*)
|    +--- org.junit:junit-bom:5.9.0 -> 5.9.1 (*)
|    +--- org.junit.platform:junit-platform-engine -> 1.9.1
|    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    +--- org.opentest4j:opentest4j:1.2.0
|    |    +--- org.junit.platform:junit-platform-commons:1.9.1 (*)
|    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    \--- org.hamcrest:hamcrest:2.2
\--- org.spockframework:spock-spring:2.3-groovy-4.0
     \--- org.spockframework:spock-core:2.3-groovy-4.0 (*)

testCompileOnly - Compile only dependencies for source set 'test'. (n)
No dependencies

testImplementation - Implementation only dependencies for source set 'test'. (n)
+--- org.springframework.boot:spring-boot-starter-test (n)
+--- org.apache.groovy:groovy:4.0.6 (n)
+--- org.apache.ivy:ivy:2.5.1 (n)
+--- org.spockframework:spock-core:2.3-groovy-4.0 (n)
\--- org.spockframework:spock-spring:2.3-groovy-4.0 (n)

testResultsElementsForTest - Directory containing binary results of running tests for the test Test Suite's test target. (n)
No dependencies

testRuntimeClasspath - Runtime classpath of source set 'test'.
+--- org.springframework.boot:spring-boot-starter-web -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0
|    |    |    +--- org.springframework:spring-core:6.0.2
|    |    |    |    \--- org.springframework:spring-jcl:6.0.2
|    |    |    \--- org.springframework:spring-context:6.0.2
|    |    |         +--- org.springframework:spring-aop:6.0.2
|    |    |         |    +--- org.springframework:spring-beans:6.0.2
|    |    |         |    |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         |    \--- org.springframework:spring-core:6.0.2 (*)
|    |    |         +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |         +--- org.springframework:spring-core:6.0.2 (*)
|    |    |         \--- org.springframework:spring-expression:6.0.2
|    |    |              \--- org.springframework:spring-core:6.0.2 (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:3.0.0
|    |    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:3.0.0
|    |    |    +--- ch.qos.logback:logback-classic:1.4.5
|    |    |    |    +--- ch.qos.logback:logback-core:1.4.5
|    |    |    |    \--- org.slf4j:slf4j-api:2.0.4
|    |    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.19.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.36 -> 2.0.4
|    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.19.0
|    |    |    \--- org.slf4j:jul-to-slf4j:2.0.4
|    |    |         \--- org.slf4j:slf4j-api:2.0.4
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    \--- org.yaml:snakeyaml:1.33
|    +--- org.springframework.boot:spring-boot-starter-json:3.0.0
|    |    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    |    +--- org.springframework:spring-web:6.0.2
|    |    |    +--- org.springframework:spring-beans:6.0.2 (*)
|    |    |    +--- org.springframework:spring-core:6.0.2 (*)
|    |    |    \--- io.micrometer:micrometer-observation:1.10.0 -> 1.10.2
|    |    |         \--- io.micrometer:micrometer-commons:1.10.2
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.1
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.14.1
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.14.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.14.1 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.14.1 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:3.0.0
|    |    +--- jakarta.annotation:jakarta.annotation-api:2.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:10.1.1
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.1
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:10.1.1
|    +--- org.springframework:spring-web:6.0.2 (*)
|    \--- org.springframework:spring-webmvc:6.0.2
|         +--- org.springframework:spring-aop:6.0.2 (*)
|         +--- org.springframework:spring-beans:6.0.2 (*)
|         +--- org.springframework:spring-context:6.0.2 (*)
|         +--- org.springframework:spring-core:6.0.2 (*)
|         +--- org.springframework:spring-expression:6.0.2 (*)
|         \--- org.springframework:spring-web:6.0.2 (*)
+--- org.springframework.boot:spring-boot-starter -> 3.0.0 (*)
+--- org.springframework.boot:spring-boot-starter-test -> 3.0.0
|    +--- org.springframework.boot:spring-boot-starter:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test:3.0.0
|    |    \--- org.springframework.boot:spring-boot:3.0.0 (*)
|    +--- org.springframework.boot:spring-boot-test-autoconfigure:3.0.0
|    |    +--- org.springframework.boot:spring-boot:3.0.0 (*)
|    |    +--- org.springframework.boot:spring-boot-test:3.0.0 (*)
|    |    \--- org.springframework.boot:spring-boot-autoconfigure:3.0.0 (*)
|    +--- com.jayway.jsonpath:json-path:2.7.0
|    |    +--- net.minidev:json-smart:2.4.7 -> 2.4.8
|    |    |    \--- net.minidev:accessors-smart:2.4.8
|    |    |         \--- org.ow2.asm:asm:9.1
|    |    \--- org.slf4j:slf4j-api:1.7.33 -> 2.0.4
|    +--- jakarta.xml.bind:jakarta.xml.bind-api:4.0.0
|    |    \--- jakarta.activation:jakarta.activation-api:2.1.0
|    +--- org.assertj:assertj-core:3.23.1
|    |    \--- net.bytebuddy:byte-buddy:1.12.10 -> 1.12.19
|    +--- org.hamcrest:hamcrest:2.2
|    +--- org.junit.jupiter:junit-jupiter:5.9.1
|    |    +--- org.junit:junit-bom:5.9.1
|    |    |    +--- org.junit.jupiter:junit-jupiter:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-engine:5.9.1 (c)
|    |    |    +--- org.junit.jupiter:junit-jupiter-params:5.9.1 (c)
|    |    |    +--- org.junit.platform:junit-platform-engine:1.9.1 (c)
|    |    |    \--- org.junit.platform:junit-platform-commons:1.9.1 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-api:5.9.1
|    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    +--- org.opentest4j:opentest4j:1.2.0
|    |    |    \--- org.junit.platform:junit-platform-commons:1.9.1
|    |    |         \--- org.junit:junit-bom:5.9.1 (*)
|    |    +--- org.junit.jupiter:junit-jupiter-params:5.9.1
|    |    |    +--- org.junit:junit-bom:5.9.1 (*)
|    |    |    \--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    |    \--- org.junit.jupiter:junit-jupiter-engine:5.9.1
|    |         +--- org.junit:junit-bom:5.9.1 (*)
|    |         +--- org.junit.platform:junit-platform-engine:1.9.1
|    |         |    +--- org.junit:junit-bom:5.9.1 (*)
|    |         |    +--- org.opentest4j:opentest4j:1.2.0
|    |         |    \--- org.junit.platform:junit-platform-commons:1.9.1 (*)
|    |         \--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    +--- org.mockito:mockito-core:4.8.1
|    |    +--- net.bytebuddy:byte-buddy:1.12.16 -> 1.12.19
|    |    +--- net.bytebuddy:byte-buddy-agent:1.12.16 -> 1.12.19
|    |    \--- org.objenesis:objenesis:3.2
|    +--- org.mockito:mockito-junit-jupiter:4.8.1
|    |    +--- org.mockito:mockito-core:4.8.1 (*)
|    |    \--- org.junit.jupiter:junit-jupiter-api:5.9.1 (*)
|    +--- org.skyscreamer:jsonassert:1.5.1
|    |    \--- com.vaadin.external.google:android-json:0.0.20131108.vaadin1
|    +--- org.springframework:spring-core:6.0.2 (*)
|    +--- org.springframework:spring-test:6.0.2
|    |    \--- org.springframework:spring-core:6.0.2 (*)
|    \--- org.xmlunit:xmlunit-core:2.9.0
+--- org.apache.groovy:groovy:4.0.6
|    \--- org.apache.groovy:groovy-bom:4.0.6
|         \--- org.apache.groovy:groovy:4.0.6 (c)
+--- org.apache.ivy:ivy:2.5.1
+--- org.spockframework:spock-core:2.3-groovy-4.0
|    +--- org.apache.groovy:groovy:4.0.4 -> 4.0.6 (*)
|    +--- org.junit:junit-bom:5.9.0 -> 5.9.1 (*)
|    +--- org.junit.platform:junit-platform-engine -> 1.9.1 (*)
|    \--- org.hamcrest:hamcrest:2.2
\--- org.spockframework:spock-spring:2.3-groovy-4.0
     +--- org.apache.groovy:groovy:4.0.4 -> 4.0.6 (*)
     \--- org.spockframework:spock-core:2.3-groovy-4.0 (*)

testRuntimeOnly - Runtime only dependencies for source set 'test'. (n)
No dependencies

(c) - dependency constraint
(*) - dependencies omitted (listed previously)

(n) - Not resolved (configuration is not meant to be resolved)

A web-based, searchable dependency report is available by adding the --scan option.

BUILD SUCCESSFUL in 3s
1 actionable task: 1 executed
```

</details>

example code:

[demo12copy.zip](https://github.com/gradle/gradle/files/10128732/demo12copy.zip)

**Comment by eskatos** [comment]
Thank you for your interest in Gradle!

Your issue appears to be a question about how to use Gradle. Your question might be answered in Gradle documentation or on our community Slack or forums. Find links to these resources on [help.gradle.org](https://help.gradle.org/).

If you think you have come across a Gradle bug, please file a new issue describing the problem and how to reproduce it.  

If you think Gradle is missing a feature, please file a new issue describing your use case for the feature.

***

You need to make sure none of the resolved configurations include ivy 2.5.0.
It seems option 4 should work though.
Please ask in our community Slack or forums.

**Comment by bademux** [comment]
@eskatos N4 is not working.
@cxjava did you found the solution?

**Comment by cxjava** [comment]
@bademux  Thank you for the update, I don't find the solution. 
I will update the latest groovy version and test it. Thank you!
